src/Features/Core/Portable/Diagnostics/EngineV1/DiagnosticIncrementalAnalyzer.cs

```diff
@@ -89,23 +111,38 @@ public override Task DocumentResetAsync(Document document, CancellationToken can
             {
                 // clear states for re-analysis and raise events about it. otherwise, some states might not updated on re-analysis
                 // due to our build-live de-duplication logic where we put all state in Documents state.
-                return ClearOnlyDocumentStates(document, raiseEvent: true, cancellationToken: cancellationToken);
+                return ClearOnlyDocumentStates(document);
             }
         }
 
-        private Task ClearOnlyDocumentStates(Document document, bool raiseEvent, CancellationToken cancellationToken)
+        private Task ClearOnlyDocumentStates(Document document)
         {
+            // since managing states and raising events are separated, it can't be cancelled.
+            var stateSets = _stateManager.GetStateSets(document.Project);
+
             // we remove whatever information we used to have on document open/close and re-calculate diagnostics
             // we had to do this since some diagnostic analyzer changes its behavior based on whether the document is opened or not.
             // so we can't use cached information.
-            ClearDocumentStates(document, _stateManager.GetStateSets(document.Project), raiseEvent, includeProjectState: false, cancellationToken: cancellationToken);
+
+            // clean up states
+            foreach (var stateSet in stateSets)
+            {
+                stateSet.Remove(document.Id, includeProjectState: false);
```

Comment:
because method name says "ClearOnlyDocumentStates" ? this will only clear document level states when file is asked to re-analyze.